PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java

```diff
@@ -163,25 +165,37 @@ protected void updateTelemetryMap(@NotNull Map<String, String> telemetryMap){
     }
 
     @NotNull
-    private WebApp getWebAppAccordingToConfiguration(@NotNull RunProcessHandler processHandler) throws Exception {
-        WebApp webApp;
+    private WebAppBase getDeployTargetByConfiguration(@NotNull RunProcessHandler processHandler) throws Exception {
         if (webAppSettingModel.isCreatingNew()) {
-            processHandler.setText(CREATE_WEBAPP);
-            try {
-                webApp = AzureWebAppMvpModel.getInstance().createWebApp(webAppSettingModel);
-            } catch (Exception e) {
-                processHandler.setText(STOP_DEPLOY);
-                throw new Exception(String.format(CREATE_FAILED, e.getMessage()));
-            }
-        } else {
-            webApp = AzureWebAppMvpModel.getInstance()
-                    .getWebAppById(webAppSettingModel.getSubscriptionId(), webAppSettingModel.getWebAppId());
+            return createWebApp(processHandler);
         }
+
+        final WebApp webApp = AzureWebAppMvpModel.getInstance()
+            .getWebAppById(webAppSettingModel.getSubscriptionId(), webAppSettingModel.getWebAppId());
         if (webApp == null) {
             processHandler.setText(STOP_DEPLOY);
             throw new Exception(NO_WEB_APP);
         }
-        return webApp;
+
+        if (isDeployingToSlot()) {
+            return webApp.deploymentSlots().getByName(webAppSettingModel.getSlotName());
+        } else {
+            return webApp;
+        }
+    }
+
+    private WebApp createWebApp(@NotNull RunProcessHandler processHandler) throws Exception {
+        processHandler.setText(CREATE_WEBAPP);
+        try {
+            return AzureWebAppMvpModel.getInstance().createWebApp(webAppSettingModel);
+        } catch (Exception e) {
+            processHandler.setText(STOP_DEPLOY);
+            throw new Exception(String.format(CREATE_FAILED, e.getMessage()));
+        }
+    }
+
+    private boolean isDeployingToSlot() {
+        return webAppSettingModel.getSlotName() != "production";
```

Comment:
Host a member in WebAppSettingModel instead of using the tricky string comparison.

Reply:
Will use `webAppSettingModel.isDeployingtoSlot()` instead